plugins/deep-link/examples/app/src-tauri/src/lib.rs

```diff
@@ -13,13 +13,41 @@ fn greet(name: &str) -> String {
 #[cfg_attr(mobile, tauri::mobile_entry_point)]
 pub fn run() {
     tauri::Builder::default()
+        .plugin(tauri_plugin_single_instance::init(|_app, argv, _cwd| {
+            println!("single instance triggered: {argv:?}");
+        }))
         .plugin(tauri_plugin_deep_link::init())
         .plugin(
             tauri_plugin_log::Builder::default()
                 .level(log::LevelFilter::Info)
                 .build(),
         )
         .setup(|app| {
+            // ensure deep links are registered on the system
+            // this is useful because AppImages requires additional setup to be available in the system
+            // and calling register() makes the deep links immediately available - without any user input
+            #[cfg(target_os = "linux")]
+            {
+                use tauri_plugin_deep_link::DeepLinkExt;
+
+                let schemes = app
+                    .config()
+                    .plugins
+                    .0
+                    .get("deep-link")
+                    .unwrap()
+                    .get("desktop")
+                    .unwrap()
+                    .get("schemes")
+                    .unwrap()
+                    .as_array()
+                    .unwrap();
```

Comment:
should we include a helper method on `DeepLinkExt` ? maybe the whole logic actually `app.deep_link().register_desktop()`

Reply:
excellent idea i love you, pushed